core/src/main/java/org/elasticsearch/index/seqno/LocalCheckpointTracker.java

```diff
@@ -183,26 +175,30 @@ synchronized void waitForOpsToComplete(final long seqNo) throws InterruptedExcep
     @SuppressForbidden(reason = "Object#notifyAll")
     private void updateCheckpoint() {
         assert Thread.holdsLock(this);
-        assert checkpoint < firstProcessedSeqNo + bitArraysSize - 1 :
-            "checkpoint should be below the end of the first bit set (o.w. current bit set is completed and shouldn't be there)";
-        assert getBitSetForSeqNo(checkpoint + 1) == processedSeqNo.getFirst() :
-            "checkpoint + 1 doesn't point to the first bit set (o.w. current bit set is completed and shouldn't be there)";
-        assert getBitSetForSeqNo(checkpoint + 1).get(seqNoToBitSetOffset(checkpoint + 1)) :
+        assert getBitArrayForSeqNo(checkpoint + 1).get(seqNoToBitArrayOffset(checkpoint + 1)) :
             "updateCheckpoint is called but the bit following the checkpoint is not set";
         try {
             // keep it simple for now, get the checkpoint one by one; in the future we can optimize and read words
-            FixedBitSet current = processedSeqNo.getFirst();
+            long bitArrayKey = getBitArrayKey(checkpoint);
+            FixedBitSet current = processedSeqNo.get(bitArrayKey);
```

Comment:
since this is a `FixedBitSet` I wonder why renamed everything to use bitArray? I don't mind. Just curious.

Reply:
Because I did not like the inconsistency that in some places it's referred to as a "bit array", and in some it's referred to as a "bit set". I wanted to make them all "bit set" but the problem is the setting `index.seq_no.checkpoint.bit_arrays_size` (currently never released to the world) refers to "bit array". If, and only if, you would be okay with changing the name of this setting to `index.seq_no.checkpoint.bit_sets_size` (in 6.0) then I would be okay with normalizing everything to "bit set".

Reply:
I'm fine with renaming the setting. I *think* I added it just be able to control things from tests and as an extra escape hatch if the default size proved wrong.  We might want to just remove it. We can solve the test part differently and I'm less concerned now about the default size (I'm also thinking that 1024 is maybe too small)..

Reply:
I opened #27191.